### PR TITLE
 pin k8s versions with integration tests 

### LINKED ISSUE #7112: Run install tests on our minimum supported k8s version
We should test that a basic installation works on our minimum supported Kubernetes version so that if we introduce backwards-incompatible changes, we can be intentional about mitigating that or changing the minimum supported version

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes #7112

This pins the default integration tests to run on the latest
k3d version by retrieving the latest k3d binary, while also
pinning `helm-deep` to run on `v1.20+` (our min k8s version). T
his way we have integration tests running on both sides
(i.e latest and minimum)

Changes include:

- Update `./bin/k3d` to always retrieve the latest k3d binary and
  use it.
- Update `_test-helpers.sh` to use min version image for `helm-deep`.
  here, we also replace flags that are removed and replaced
  with newer ones with the latest k3d binary.

I've tried to find a GitHub action for K3s, but most of them seem to
be have a had a release while ago, and not goes well with our plan of
somehow getting the latest k8s.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

**Comment by olix0r** [comment]
integration test failures when loading images:

```
ERRO[0010] failed to import images in node 'k3d-helm-deep-server-0': docker failed to create exec config for node 'k3d-helm-deep-server-0': Error response from daemon: Container 8771c4151174fdf1dd9ff5c50939411154617172d36ff2e6392503ad062356bc is restarting, wait until the container is running 
50
ERRO[0018] failed to import images in node 'k3d-helm-deep-server-0': Exec process in node 'k3d-helm-deep-server-0' failed with exit code '137'
```

Also note that the latest k3d release supports specifying logical versions like `+v1.20` per https://github.com/rancher/k3d/releases/tag/v5.1.0

**Comment by Pothulapati** [comment]
Updated this PR, to use the latest `k3d` binary **always** but use an older image version with `helm-deep` as some flags are replaced btw binaries, and it's hard to maintain them based on version.

**Review by alpeb — COMMENTED** [review]
Changes look good, but for some reason the `helm-deep` integration test is still show `v1.21`:
https://github.com/linkerd/linkerd2/runs/4163956319?check_suite_focus=true#step:8:32

**Comment by Pothulapati** [comment]
@alpeb Yep, That is weird and I think that version is about the k3s wrapper. The k8s version seems to be the expected `v1.20+` https://github.com/linkerd/linkerd2/runs/4163956319?check_suite_focus=true#step:8:61

**Review by alpeb — APPROVED** [review]

File: bin/k3d [hunk 1]
```diff
@@ -2,11 +2,9 @@
 
 set -eu
 
-k3dversion=v4.4.5
```
Comment (kleimkuhler) [inline bin/k3d:5]: 👍

**Review by kleimkuhler — APPROVED** [review]
Nice!

**Pull request merged**